src/OpenCensus.Exporter.ApplicationInsights/Implementation/TraceExporterHandler.cs

```diff
@@ -38,38 +39,158 @@ public void Export(IList<ISpanData> spanDataList)
+
+                string data = null;
+                string target = null;
+                string type = null;
+                string userAgent = null;
+
+                IAttributeValue spanKindAttr = null;
+                IAttributeValue errorAttr = null;
+                IAttributeValue httpStatusCodeAttr = null;
+                IAttributeValue httpMethodAttr = null;
+                IAttributeValue httpPathAttr = null;
+                IAttributeValue httpHostAttr = null;
+                IAttributeValue httpUrlAttr = null;
+                IAttributeValue httpUserAgentAttr = null;
+                IAttributeValue httpRouteAttr = null;
+                IAttributeValue httpPortAttr = null;
+
+                foreach (var attr in span.Attributes.AttributeMap)
+                {
+                    var key = attr.Key;
+                    if (key == "span.kind")
+                    {
+                        spanKindAttr = attr.Value;
+                    }
+                    else if (key == "error")
+                    {
+                        errorAttr = attr.Value;
+                    }
+                    else if (key == "http.method")
+                    {
+                        httpMethodAttr = attr.Value;
+                    }
+                    else if (key == "http.path")
+                    {
+                        httpPathAttr = attr.Value;
+                    }
+                    else if (key == "http.host")
+                    {
+                        httpHostAttr = attr.Value;
+                    }
+                    else if (key == "http.url")
+                    {
+                        httpUrlAttr = attr.Value;
+                    }
+                    else if (key == "http.status_code")
+                    {
+                        httpStatusCodeAttr = attr.Value;
+                    }
+                    else if (key == "http.user_agent")
+                    {
+                        httpUserAgentAttr = attr.Value;
+                    }
+                    else if (key == "http.route")
+                    {
+                        httpRouteAttr = attr.Value;
+                    }
+                    else if (key == "http.port")
+                    {
+                        httpPortAttr = attr.Value;
+                    }
+                    else
+                    {
+                        var value = attr.Value.Match<string>(
+                            (s) => { return s; },
+                            (b) => { return b.ToString(); },
+                            (l) => { return l.ToString(); },
+                            (d) => { return d.ToString(); },
+                            (obj) => { return obj.ToString(); });
+
+                        props.Add(attr.Key, value);
+                    }
+                }
 
-                if (span.Kind == SpanKind.Server)
+                this.OverwriteSpanKindFromAttribute(spanKindAttr, ref resultKind);
```

Comment:
most likely it's never needed. How about we do it only if there is no span kind set on the span?